src/util/psbt/map/output.rs

```diff
@@ -92,33 +93,67 @@ impl PartialEq for TapTree {
 
 impl Eq for TapTree {}
 
+impl AsRef<TaprootBuilder> for TapTree {
+    #[inline]
+    fn as_ref(&self) -> &TaprootBuilder {
+        &self.0
+    }
+}
+
+impl Borrow<TaprootBuilder> for TapTree {
+    #[inline]
+    fn borrow(&self) -> &TaprootBuilder {
+        &self.0
+    }
+}
+
+impl From<TapTree> for TaprootBuilder {
+    #[inline]
+    fn from(tree: TapTree) -> Self {
+        tree.into_builder()
+    }
+}
+
 impl TapTree {
+    /// Convenience method to retrieve number of script leafs in the tree.
+    #[inline]
+    pub fn scripts_count(&self) -> usize {
+        self.0.branch().len()
+    }
+
     /// Gets the inner node info as the builder is finalized.
-    fn node_info(&self) -> &NodeInfo {
+    pub fn node_info(&self) -> &NodeInfo {
         // The builder algorithm invariant guarantees that is_complete builder
         // have only 1 element in branch and that is not None.
         // We make sure that we only allow is_complete builders via the from_inner
         // constructor
         self.0.branch()[0].as_ref().expect("from_inner only parses is_complete builders")
     }
 
-    /// Converts a [`TaprootBuilder`] into a tree if it is complete binary tree.
+    /// Constructs [`TapTree`] from a [`TaprootBuilder`] if it is complete binary tree.
     ///
     /// # Return
-    /// A `TapTree` iff the `inner` builder is complete, otherwise return the inner as `Err`.
-    pub fn from_inner(inner: TaprootBuilder) -> Result<Self, TaprootBuilder> {
-        if inner.is_complete() {
-            Ok(TapTree(inner))
+    /// A `TapTree` iff the `builder` builder is complete, otherwise return the inner as `Err`.
```

Comment:
```suggestion
    /// A `TapTree` iff the `builder` is complete, otherwise return the `builder` as `Err`.
```

Not part of this PR but the return section header should be plural i.e., `# Returns`.

Reply:
Fixed elsewhere: https://github.com/rust-bitcoin/rust-bitcoin/pull/929/files#diff-8ed6474e7f75e74be2ab10a3102be1af7dcda692d83edceff3012296e3a7e04eR140